utils/python/CIME/case.py

```diff
@@ -128,13 +128,10 @@ def initialize_derived_attributes(self):
         self.tasks_per_numa = int(math.ceil(self.tasks_per_node / 2.0))
         smt_factor = max(1,int(self.get_value("MAX_TASKS_PER_NODE")/self.get_value("PES_PER_NODE")))
 
-        self.cores_per_task = ((self.get_value("MAX_TASKS_PER_NODE")/smt_factor) \
-                               / self.tasks_per_node) * 2
+        self.cores_per_task = self.thread_count / smt_factor
```

Comment:
This is currently only used on cori.   The new calculation makes a lot more sense for the variable name than the old one did and I think nersc has suggested that we don't need this field on cori anyway.  I'm testing cori (both knl and haswell) now with that field removed and will report results tomorrow.